web/html/src/manager/recurring/recurring-actions-details.tsx

```diff
@@ -0,0 +1,276 @@
+import * as React from "react";
+
+import _sortBy from "lodash/sortBy";
+
+import { Button } from "components/buttons";
+import { DeleteDialog } from "components/dialog/DeleteDialog";
+import { ModalButton } from "components/dialog/ModalButton";
+import { Messages, Utils as MessagesUtils } from "components/messages";
+import { BootstrapPanel } from "components/panels/BootstrapPanel";
+import { TopPanel } from "components/panels/TopPanel";
+import { Column } from "components/table/Column";
+import { Table } from "components/table/Table";
+
+import Network from "utils/network";
+
+import { DisplayHighstate } from "../state/display-highstate";
+import { isReadOnly, targetNameLink, targetTypeToString } from "./recurring-actions-utils";
+
+function channelIcon(channel) {
+  let iconClass, iconTitle, iconStyle;
+  if (channel.type === "state") {
+    iconClass = "fa spacewalk-icon-salt-add";
+    iconTitle = t("State Configuration Channel");
+  } else if (channel.type === "internal_state") {
+    iconClass = "fa spacewalk-icon-salt-add";
+    iconTitle = t("Internal State");
+    iconStyle = { border: "1px solid black" };
+  } else {
+    iconClass = "fa spacewalk-icon-software-channels";
+    iconTitle = t("Normal Configuration Channel");
+  }
+
+  return <i className={iconClass} title={iconTitle} style={iconStyle} />;
+}
+
+type RecurringActionsDetailsProps = {
+  data?: any;
+  minions?: any;
+  onCancel: (arg0: string) => any;
+  onEdit: (arg0: any) => any;
+  onError: (arg0: any) => any;
+  onDeleteError: (arg0: any) => any;
+  onSetMessages: (arg0: any) => any;
+};
+
+type RecurringActionsDetailsState = {
+  messages: any[];
+  minions: any;
+  details: any;
+};
+
+class RecurringActionsDetails extends React.Component<RecurringActionsDetailsProps, RecurringActionsDetailsState> {
+  weekDays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
```

Comment:
This is ok for getting the feature out right away, but we will want to refactor this out later.

Reply:
What do you have in mind? A calendar library? or do we already have something for that?